draft.md

```diff
@@ -84,14 +84,44 @@ The idempotency key that is supplied as part of every `POST` request MUST be uni
 
 How to make the key unique is up to the client and it's agreed protocol with the server. It is `RECOMMENDED` that `UUID` or a similar random identifier be used as the idempotency key.
 
+## Idempotency Key Validity and Expiry
 
-## Resource Behavior
+The server MAY enforce a time based idempotency keys, thus, be able to purge or delete a key upon its expiry. Resource SHOULD publish expiration policy related documentation.
 
-How an idempotency key is made unique key is determined by the client. We `RECOMMEND` using `UUID` or similar random string with enough entropy to avoid collisions.
+## Idempotency Fingerprint
 
-The resource SHOULD save the resulting `HTTP` status code and body of the first request made for any given idempotency key, regardless of whether it succeeded or failed. Subsequent requests with the same key `MUST` return the same result, including errors.
+An idempotency fingerprint MAY be used in conjunction with with an idempotency key to determine the uniqueness of a request and is generated from request payload data. An idempotency fingerprint is generated by the server implementation and MAY employ any of the following patterns.
 
-It is also recommended that the resource implements the idempotency keys to be time-based and, thus, be able to purge or delete a key upon its expiry. Resource SHOULD publish expiration policy related documentation.
+        (A) Checksum of the entire request payload.
+        (B) Checksum of selected elements in the request payload.
+        (C) Field value match for each field in the request payload.
+        (D) Field value match for selected elements in the request payload.
+        (E) Request digest/signature.
+
+## Idempotency Enforcement Scenarios
+
+First time Request (Idempotency Key and Idempotency FIngerprint scenarios has not been seen)
+
+    The server SHOULD process the request normally and the server responds with an appropriate response and status code.
+
+Duplicate Request (Idempotency Key and Idempotency Fingerprint scenarios has been seen)
+
+    Replay
+        The request was replayed after the original request completed. The server MUST respond with the result of the previously completed operation, success or an    error
+        
+    Concurrent Request
+        The request was replayed before the original request completed. The server MUST respond with a resource conflict error. See #Error scenarions for details.
+       
+## Responsibilities
+
+    Client
+      For the idempotent resource operations, the client MUST present a unique idempotency key as part of the request.
+      
+    Server
+      Generate Idempotency Fingerprint when required.
+      Enforce compliance with the Idempotency enforcement scenarios.
```

Comment:
Check for idempotency under various scenarios including the ones described earlier.